### PR TITLE
migrate csvmonkey to conan v2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
This conan v2 migration PR was generated by automatic script version 6.
It uses simple find and replace technics. Feel free to extend this PR.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/csvmonkey/all/test_package/conanfile.py [hunk 1]
```diff
@@ -11,6 +12,6 @@ def build(self):
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
+        if not tools.build.cross_building(self, self):
```
Comment (ericLemanissier) [inline recipes/csvmonkey/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.build.cross_building(self):
```
again

File: recipes/csvmonkey/all/conanfile.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import os
 
-from conans.errors import ConanInvalidConfiguration
-from conans import ConanFile, CMake, tools
+from conan.errors import ConanInvalidConfiguration
+from conan import ConanFile, tools
+from conans import CMake
```
Comment (ericLemanissier) [inline recipes/csvmonkey/all/conanfile.py:5]:
```suggestion
```

Comment (AndreyMlashkin) [reply recipes/csvmonkey/all/conanfile.py:5]: It will lead to a fail. Not everything can be automated

Comment (ericLemanissier) [reply recipes/csvmonkey/all/conanfile.py:5]: it won't fail, cmake is not used in the recipe.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

**Comment by uilianries** [comment]
I just re-triggered right now.

**Review by uilianries — APPROVED** [review]

**Pull request merged**